### ISSUE TITLE
Openstack Swift (Amazon S3) - Listing directory failed

### ISSUE THREAD

**Issue by zoomer85** [iterate-ch/cyberduck#13383]
**Describe the bug**
Hello, when i try to use Openstack swift over Amazon S3, i get an error when try to listing "bucket".
The behavior is absolutely similar on several providers.
Have been tested: OVH, Oracle, SC

When trying to list the contents of the container, an error occurs and nothing else happens.
`"Listing directory XXX failed. Failed to parse XML document with handler class org.jets3t.service.impl.rest.XmlResponsesSaxParser$ListBucketHandler. Please contact your web hosting service provider for assistance."
`
**To Reproduce**
To repeat, you can use the object storage service of any provider that provides Openstack Swift through S3.

**Expected behavior**
I expected to see a listing of files in a container

****Cyberduck Version 8.3.3 (37544) (MacOS)****

**Comment by zoomer85** [comment]
Very similar to a bug that may have been previously fixed, but is now back again.
https://github.com/iterate-ch/cyberduck/issues/10471

Now i have next reply with s3cmd:
```
DEBUG: Response:
{'data': b"<?xml version='1.0' encoding='UTF-8'?>\n<ListBucketResult xml"
         b'ns="http://s3.amazonaws.com/doc/2006-03-01/"><Name>bucketname</Name><Prefix'
         b'/><Marker/><MaxKeys>1000</MaxKeys><Delimiter>/</Delimiter><IsTruncat'
         b'ed>false</IsTruncated><Contents><Key>Screenshot 2022-04-14 at 23.50.'
         b'21.png</Key><LastModified>2022-05-12T13:40:17.541Z</LastModified><ET'
         b'ag>"3fcf1189802cc75006100190372048ad"</ETag><Size>187976</Size><Owne'
         b'r><ID>testtenant:testuser</ID><DisplayName>testtenant:testuser</DisplayName></'
         b'Owner><StorageClass>STANDARD</StorageClass></Contents><Contents><Key'
         b'>Screenshot 2022-04-14 at 23.50.36.png</Key><LastModified>2022-05-12'
         b'T13:40:17.541Z</LastModified><ETag>"5e3fedc3bf0fd8e3aead5cecaa285fdb'
         b'"</ETag><Size>137389</Size><Owner><ID>testtenant:testuser</ID><DisplayNam'
         b'e>testtenant:testuser</DisplayName></Owner><StorageClass>STANDARD</Storag'
         b'eClass></Contents></ListBucketResult>',
 'headers': {'connection': 'keep-alive',
             'content-length': '845',
             'content-type': 'application/xml',
             'date': 'Fri, 13 May 2022 10:22:51 GMT',
             'server': 'nginx',
             'x-amz-id-2': 'txd6d39ecaa69c4078b7074-00627e317b',
             'x-amz-request-id': 'txd6d39ecaa69c4078b7074-00627e317b',
             'x-openstack-request-id': 'txd6d39ecaa69c4078b7074-00627e317b',
             'x-trans-id': 'txd6d39ecaa69c4078b7074-00627e317b'},
 'reason': 'OK',
 'status': 200}
```

Clean XML

```xml
<?xml version='1.0' encoding='UTF-8'?>
<ListBucketResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
    <Name>bucketname</Name>
    <Prefix/>
    <Marker/>
    <MaxKeys>1000</MaxKeys>
    <Delimiter>/</Delimiter>
    <IsTruncated>false</IsTruncated>
    <Contents>
        <Key>Screenshot 2022-04-14 at 23.50.21.png</Key>
        <LastModified>2022-05-12T13:40:17.541Z</LastModified>
        <ETag>"3fcf1189802cc75006100190372048ad"</ETag>
        <Size>187976</Size>
        <Owner>
            <ID>testtenant:testuser</ID>
            <DisplayName>testtenant:testuser</DisplayName>
        </Owner>
        <StorageClass>STANDARD</StorageClass>
    </Contents>
    <Contents>
        <Key>Screenshot 2022-04-14 at 23.50.36.png</Key>
        <LastModified>2022-05-12T13:40:17.541Z</LastModified>
        <ETag>"5e3fedc3bf0fd8e3aead5cecaa285fdb"</ETag>
        <Size>137389</Size>
        <Owner>
            <ID>testtenant:testuser</ID>
            <DisplayName>testtenant:testuser</DisplayName>
        </Owner>
        <StorageClass>STANDARD</StorageClass>
    </Contents>
</ListBucketResult>'
```


Just like that time, the element `Version` is not available.

**Comment by clementnuss** [comment]
just for info, I'm reporting the same behaviour with MacOS version 8.3.3 and trying to connect to Infomaniak SwissBackup (also an Openstack Swift S3 mode):
<img width="284" alt="image" src="https://user-images.githubusercontent.com/2116997/169355552-b34592dc-f5eb-4e47-9bcf-c72a77f8ecc4.png">

**Comment by dkocher** [comment]
> just for info, I'm reporting the same behaviour with MacOS version 8.3.3 and trying to connect to Infomaniak SwissBackup (also an Openstack Swift S3 mode): <img alt="image" width="284" src="https://user-images.githubusercontent.com/2116997/169355552-b34592dc-f5eb-4e47-9bcf-c72a77f8ecc4.png">

As a workaround please try to connect using the OpenStack Swift connection profile from _Preferences → Profiles_.

**Comment by dkocher** [comment]
> Very similar to a bug that may have been previously fixed, but is now back again. #10471

I cannot seem to get a test failure in e84e78f42a444f912e005c674591fc7bb311120c

**Comment by dkocher** [comment]
> The behavior is absolutely similar on several providers. Have been tested: OVH, Oracle, SC

As a workaround please try to connect using the OpenStack Swift connection profiles for OVH and S3 connection profiles for OCI Object Storage from _Preferences → Profiles_.

**Comment by zoomer85** [comment]
Hello,

i am try with "OCI Object Storage" and get next error

```
Listing directory / failed.
The request signature we calculated does not match the signature you provided. Check your key and signing method. Please contact your web hosting service provider for assistance.
```

<img width="269" alt="Screenshot 2022-05-20 at 12 52 00" src="https://user-images.githubusercontent.com/5430542/169503714-30b200b7-3347-475a-9b3d-d714b10af6f1.png">

**Comment by zoomer85** [comment]
Later i am test some profiles and found working profile with Openstack Swift over S3 - "S3 (Deprecated path style requests)".

Please create one more profile with name "Openstack Swift over S3", or something similar. Or add comment to "S3 (Deprecated path style requests)".

**Comment by clementnuss** [comment]
hi,

in my case, with OCI Object Storage (Oracle logo), I changed the server name and it works perfectly for infomaniak SwissBackup S3, thanks for the Workaround!

it would be nice to have a more generic profile for problematic S3 shares, or a tickbox "compatibility mode" for the "Amazon S3" to permit using 3rd party S3 providers.

thanks for the workaround already, really helping!

**Comment by dkocher** [comment]
Fix error reporting to get exact parser failure in #13405.

**Comment by clementnuss** [comment]
@dkocher I think we need to reopen this issue, with `8.4.2` the error is still there, and the parser doesn't show much more info:

<img width="270" alt="image" src="https://user-images.githubusercontent.com/2116997/181682243-942e47b9-25e5-42e2-bc94-5d46cfa4247f.png">

also, for whatever reason, the workaround using _OCI Object Storage_ also produces the same error now.

If you prefer, I could also reopen another issue

**Comment by dkocher** [comment]
> @dkocher I think we need to reopen this issue, with `8.4.2` the error is still there, and the parser doesn't show much more info:

Can you confirm you are using one of the _Infomaniak Swiss Backup_ connection profiles?

**Comment by dkocher** [comment]
> > @dkocher I think we need to reopen this issue, with `8.4.2` the error is still there, and the parser doesn't show much more info:
> 
> Can you confirm you are using one of the _Infomaniak Swiss Backup_ connection profiles?

Connection Profiles for _Infomaniak Swiss Backup_ have been made available in e4bc1eba and #13468.

**Comment by clementnuss** [comment]
I'm not using SwissBackup profiles, as those are made for the Swift API of Infomaniak, and expect the following credentials:
<img width="530" alt="image" src="https://user-images.githubusercontent.com/2116997/182677813-a671e37b-2f62-40fb-9718-9e5a87f5b976.png">

In my case however, I'm using SwissBackup in S3 mode, for which the endpoint is `s3.swiss-backup02.infomaniak.com` and it uses the S3 protocol.

with the previous version of Cyberduck and the Oracle OCI S3 profile, I was able to work with SwissBackup S3 mode.
With the current version however, the error is again there.

I can open a new issue if you prefer?
